docs/compilers/Design/Editor Config.md

```diff
@@ -0,0 +1,25 @@
+.editorconfig Compiler Support
+==============================
+
+Linked or "External" Files
+==========================
+
+The compiler has no notion of whether a source file is linked or brought in via some external source (e.g. NuGet package). This design is preserved for .editorconfig, where the only mechanism to apply editorconfig files to external files is to map or copy the files into an appropriate location in the filesystem, and then include that location in the pathspec for the editorconfig. In the case of a linked file, this can often be done via a relative path. For instance, say the code layout looks like the following:
+
+```
+Repo Root
+  | --- LinkedFile.cs
+  | --- Subproject
+          | --- .editorconfig
+          | --- SourceFile1.cs
+          | --- SourceFile2.cs
+```
+
+In this example the subproject's `.editorconfig` will not apply to `LinkedFile.cs` if it only has the specification `[*.cs]`. If the user wants to explicitly map in the linked file such that the subproject's editorconfig applies, the `.editorconfig` file would need to have a new section added that points to the linked file with a relative path specification. For instance,
+
+```
+[../LinkedFile.cs]
+option = value
+```
+
+For external files, like those provided by source NuGet packages, the NuGet package restore directory can itself be mapped into the repository path. Then, the relative path can be added to the `.editorconfig` like for linked files.
```

Comment:
I'd add: 

To enable application of `.editorconfig` rules to source packages referenced by a project in your repository set the `NuGetPackageRoot` project property to a directory under the repository root.

> We might consider adding support for source packages restored to a NuGet cache directory located outside of the repository root. This could be achieved like so: MSBuild/NuGet would create hard-links (or copies) in the repository root to each source package directory based on the information available in [source roots](https://github.com/tmat/repository-info/tree/master/docs#sourceroot). For example, `.sourcepackages/{SourcePackageName}`. Then root `.editorconfig` could use pathspecs like `.sourcepackages/{SourcePackageName}/**/*.cs`.

Reply:
I don't want to give specific guidance on how to set the NuGet package restore directory. Property is one way, environment variable is another way -- this is something the NuGet docs should make clear and we can leave it up to the user.